### DISCUSSION TITLE
Keyword arguments in model instance methods?

### DISCUSSION THREAD

**Discussion by destroytoday** [jeremyevans/sequel#2021] [General]
Hi! Been using Sequel in my app for over 9 years now and _really_ appreciate all the work that went into it.

I’m in the process of upgrading from Ruby 2.7 to 3.1 and in doing so, I hit an issue with how Sequel models handle keyword arguments. I’m on the latest Sequel version, but narrowed it down to keyword arguments being passed to a model’s method being treated like a single argument hash, so I get the classic `wrong number of arguments (given 1, expected 0)`

Here’s what the code sorta looks like:

```ruby
class InvoiceEmail < Sequel::Model
  def deliver(contacts: [], bcc: false)
    # code that would reference `contacts` and `bcc` to deliver emails
  end
end

InvoiceEmail.first.deliver(contacts: ["jerry@seinfeld.com"], bcc: true)
```

In Ruby 2.7, this worked no problem, and I assumed by updating the gem that it would work with the latest Ruby 3 keyword arguments, but I now get this error. If I change it to `def deliver(options)`, I can reference them like `options[:contacts]`, but I’d much rather be able to reference them directly if possible.

Any pointers would be incredibly welcome!

**Comment by jeremyevans** [comment]
There must be more code than you are showing (such a `deliver` getting wrapped), because the code as displayed works fine.  Can you put together a minimal self-contained reproducible example showing the problem?

**Reply by destroytoday** [reply]
Good call! I used TracePoint to inspect the call path and found that a slightly older version of Rspec was indeed wrapping the method, so updated to the latest version and it works as expected. Thanks so much for the hint!
